requre/cassette.py

```diff
@@ -21,14 +21,670 @@
 # SOFTWARE.
 from pathlib import Path
 
-from requre.storage import PersistentObjectStorage
+import logging
+from typing import Dict, Optional, List, Hashable, Any, Callable
+from enum import Enum
+import os
+import yaml
+import time
+import inspect
 
+from requre.utils import StorageMode
+from requre.constants import (
+    METATADA_KEY,
+    ENV_REQURE_STORAGE_MODE,
+    ENV_STORAGE_FILE,
+    VERSION_REQURE_FILE,
+    KEY_MINIMAL_MATCH,
+)
+from requre.exceptions import (
+    PersistentStorageException,
+    ItemNotInStorage,
+    StorageNoResponseLeft,
+)
+
+# use this sleep to avoid decorating original time function used internally
+original_sleep = time.sleep
+original_time = time.time
+logger = logging.getLogger(__name__)
+
+
+class StorageKeysInspect:
+    @staticmethod
+    def get_base_keys(func: Callable) -> List[Any]:
+        raise NotImplementedError("Use child classes")
+
+
+class StorageKeysInspectFull(StorageKeysInspect):
```

Comment:
I am thinking about naming.

- What is the purpose of those classes?
  - Can you please document that in the abstract class? (And the method as well.)

Do I get it correctly?
- The `get_base_keys` method is the only method of classes.
- The `get_base_keys` method gets a function and returns a list of strings representing the logical path to that function?
- We have different ways how to get them => what about having a "strategy" in the name?

Reply:
This is still relevant to me. Or, do you want to solve that later as well?